src/com/xilinx/rapidwright/util/ReportRouteStatus.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import com.xilinx.rapidwright.design.Design;
+import com.xilinx.rapidwright.design.DesignTools;
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.design.SitePinInst;
+import com.xilinx.rapidwright.device.Node;
+import com.xilinx.rapidwright.device.PIP;
+import com.xilinx.rapidwright.rwroute.RouterHelper;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportRouteStatus {
+    public static void main(String[] args) {
+        if (args.length != 1) {
+            System.out.println("USAGE: ReportRouteStatus <design.dcp>");
+            System.exit(1);
+        }
+
+        Design design = Design.readCheckpoint(args[0]);
+
+        DesignTools.updatePinsIsRouted(design);
+
+        Map<Node, Net> nodesUsedByDesign = new HashMap<>();
+        Set<Net> conflictingNets = new HashSet<>();
+
+        Collection<Net> nets = design.getNets();
+        int numPhysicalNets = nets.size();
+        int numRoutableNets = 0;
+        int numUnroutedNets = 0;
+        int numNetsWithUnroutedPins = 0;
+        for (Net net : nets) {
+            if (!RouterHelper.isRoutableNetWithSourceSinks(net)) {
+                continue;
+            }
+            numRoutableNets++;
+
+            boolean isFullyRouted = true;
+            boolean isPartiallyRouted = false;
+            for (SitePinInst spi : net.getPins()) {
+                if (spi.isRouted()) {
+                    isPartiallyRouted = true;
+                    continue;
+                }
+                isFullyRouted = false;
+            }
+
+            boolean isConflictFree = true;
+            for (PIP pip : net.getPIPs()) {
+                Node endNode = pip.isReversed() ? pip.getStartNode() : pip.getEndNode();
+                Net conflictingNet = nodesUsedByDesign.putIfAbsent(endNode, net);
+                if (conflictingNet != null && conflictingNet != net) {
+                    conflictingNets.add(conflictingNet);
+                    isConflictFree = false;
+                }
+            }
+
+            if (!isConflictFree) {
+                conflictingNets.add(net);
+            } else if (!isFullyRouted) {
+                if (isPartiallyRouted) {
+                    numNetsWithUnroutedPins++;
+                } else {
+                    numUnroutedNets++;
+                }
+            }
+        }
+
+        int numConflictingNets = conflictingNets.size();
+        System.out.println();
+        System.out.println("Design Route Status");
```

Comment:
Should we tweak the title of the output ("RapidWright Design Route Status", for example) to make it easily distinguishable from the Vivado `report_route_status` output?  It might be nice to know if we are getting a RapidWright report vs. Vivado report.